frontend/apps/filemanager/filemanager.lua

```diff
@@ -214,9 +216,19 @@ function FileManager:setupLayout()
     end
 
     function file_chooser:onFileSelect(file)  -- luacheck: ignore
-        local ReaderUI = require("apps/reader/readerui")
-        ReaderUI:showReader(file)
-        return true
+        if FileManager.instance.select_mode then
```

Comment:
You can probably pass the FileManager instance as a property of FileChooser
` filemanager = self`
and have it replace the current `is_file_manager=true` (and in menu.lua just a check for self.filemanager instead).